tinynn/converter/operators/torch/custom_gru.py

```diff
@@ -0,0 +1,441 @@
+                            ops.append(tfl.AddOperator([input_mm_list[-1], ngate_in_h], [ngate_in]))
+
+                        gate_n = self.create_transform_tensor(np.tanh(ngate_in.tensor))  # np.tanh vs torch.tanh ?
+                        ops.append(tfl.TanhOperator([ngate_in], [gate_n])) # 齐了 rzn gate的output
+
+                        h_leftA = self.create_transform_tensor(-1 * gate_z.tensor)
+                        ops.append(tfl.MulOperator([-1, gate_z], [h_leftA]))
+                        h_leftB = self.create_transform_tensor(1 + h_leftA.tensor)
+                        ops.append(tfl.AddOperator([1, h_leftA], [h_leftB]))
+                        h_left = self.create_transform_tensor(h_leftA.tensor * h_leftB.tensor)
+                        ops.append(tfl.MulOperator([h_leftA, h_leftB], [h_left]))
+
+                        if i != 0:
+                            h_right = self.create_transform_tensor(gate_z.tensor * h.tensor)
+                            ops.append(tfl.MulOperator([gate_z, h], [h_right]))
+                        
+                        if i != 0:
+                            h = self.create_transform_tensor(h_left.tensor + h_right.tensor)
+                            ops.append(tfl.AddOperator([h_left, h_right], [h]))
+                        else:
+                            h = h_left
+
+                        stacked_hs.append(h)
+
+                    tf_out_state_tensors[0].append(h)
+
+                    output_ts.extend(stacked_hs[::stride])
+                
+                '''
+                if bidirectional:
+                    # For bidirectional LSTMs, the forward output tensors and the backward output tensors are
+                    # concatenated before we pack them together
+                    fw_out = self.create_transform_tensor(
+                        np.stack([x.tensor for x in output_ts[:num_timestep]], ts_axis)
+                    )
+                    ops.append(tfl.PackOperator(output_ts[:num_timestep], [fw_out], num_timestep, axis=ts_axis))
+
+                    bw_out = self.create_transform_tensor(
+                        np.stack([x.tensor for x in output_ts[num_timestep:]], ts_axis)
+                    )
+                    ops.append(tfl.PackOperator(output_ts[num_timestep:], [bw_out], num_timestep, axis=ts_axis))
+
+                    ops.append(tfl.ConcatenationOperator([fw_out, bw_out], outputs, axis=2))
+                '''
+                if layer_idx != num_layers - 1:
+                    # Reusing unpacked tensors for the logic in the next layer
+                    unpacked_tensors[outputs[0].name] = output_ts
+                else:
+                    # For the last layer, we have to pack the together
+                    ops.append(tfl.PackOperator(output_ts, outputs, len(output_ts), axis=ts_axis))
+
+            
+            else:  # TODO: 不展开怎么改 
```

Comment:
You'll have to add the schema of the custom operator `unidirectional_gru`, so my suggestion is to mark this as todo.